### PR TITLE
Add support for placement group

### PR DESCRIPTION
**Pull request by mkatturu**

### CODE REVIEW

File: services/compute/compute.go [hunk 1]
```diff
@@ -1225,6 +1242,46 @@ type AvailabilitySet struct {
 	VirtualMachines []*VirtualMachineReference
 }
 
+type PlacementGroup struct {
+	// ID
+	ID *string `json:"ID,omitempty"`
+	// Name
+	Name *string `json:"name,omitempty"`
+	// Type
+	Type PlacementGroupType `json:"type,omitempty"`
```
Comment (piyushka17) [inline services/compute/compute.go:1251]: just curious why this PG property is not in 'PlacementGroupProperties' struct.

**Review by piyushka17 — DISMISSED** [review]

**Review by bgriddaluru — DISMISSED** [review]

File: services/compute/placementgroup/placementgroup.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	}
+
+	if s.PlacementGroupProperties != nil {
+		if s.PlacementGroupProperties.Zones != nil {
+			if s.PlacementGroupProperties.StrictPlacement {
+				placementGroup.Zones.StrictPlacement = true
+			}
+
+			for _, zn := range *s.PlacementGroupProperties.Zones {
+				rpcZoneRef, err := getRpcZoneReference(&zn)
```
[the remark this answers is not in the document]
Comment (mkatturu) [reply services/compute/placementgroup/placementgroup.go:60]: okay will be addressed in next PR

File: services/compute/compute.go (hunk 1 above)
Comment (mkatturu) [reply services/compute/compute.go]: Moved it inside placement group properties

**Review by piyushka17 — APPROVED** [review]

**Review by bgriddaluru — APPROVED** [review]

**Comment by mkatturu** [comment]
This changes are merged via another PR: https://github.com/microsoft/moc-sdk-for-go/pull/290

**Pull request closed without merging**